app/services/index_deleter.rb

```diff
@@ -0,0 +1,37 @@
+class IndexDeleter
+  def initialize(args)
+    @index_alias = args[:index_alias]
+    @client = args[:client]
+    @logger = args[:logger]
+  end
+
+  def run
+    aliases = client.indices.get_aliases.keys
+    indexes_to_be_deleted = prepare_indexes_for_deletion aliases
+    delete indexes_to_be_deleted
+  rescue => e
+    msg = "Failed to delete alias.\n#{e.message}"
+    logger.error msg
+    Raven.capture_error msg
+  end
+
+  private
+
+  attr_reader :client, :index_alias, :logger
+
+  def prepare_indexes_for_deletion(aliases)
+    # Ensure that the three most recent indexes are kept
+    aliases
+      .select { |alias_name| alias_name.include? index_alias }
+      .sort_by { |alias_name| Time.parse(alias_name.gsub(/"#{index_alias}_"/, '')) }
+      .reverse
+      .slice(2...-1)
```

Comment:
so you're reversing the array and taking the last three? How about keeping the array as is, and then taking the three first ones with the super nice https://ruby-doc.org/core-2.2.0/Array.html#method-i-take ?

Reply:
regarding sorting, this might be overkill here, but still interesting to consider: https://ruby-doc.org/core-2.4.0/Comparable.html.

Reply:
instead of using `select`, you might also consider the very nice https://ruby-doc.org/core-2.1.0/Enumerable.html#method-i-grep.